src/Tools/ExternalAccess/CustomMessage/External/DocumentLinePosition.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.Serialization;
+using System.Text.Json.Serialization;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CustomMessageHandler;
+
+public struct DocumentLinePosition
+{
+    public DocumentLinePosition(Document document, LinePosition linePosition)
+    {
+        _ = document ?? throw new ArgumentNullException(nameof(document));
+
+        FilePath = document.FilePath ?? throw new InvalidOperationException("Missing document file path");
+        Line = linePosition.Line;
+        Character = linePosition.Character;
+    }
+
+    [JsonConstructor]
+    public DocumentLinePosition(string filePath, int line, int character)
+    {
+        FilePath = filePath ?? throw new ArgumentNullException(nameof(filePath));
+        Line = line;
+        Character = character;
+    }
+
+    [JsonPropertyName("filePath")]
+    public string FilePath { get; }
+
+    [JsonPropertyName("line")]
+    public int Line { get; }
+
+    [JsonPropertyName("character")]
+    public int Character { get; }
```

Comment:
we try to not do thigns in Line/Column in roslyn for users.  It should be just a position.

Reply:
I have no problem changing this into just a number. But, LSP uses line/character. I believe the reason for it is that, if an edit happens that we are not able to compensate for, the line/character positions result in a more intuitive displacement.
Let me know what you think is best.

Reply:
Discussed in chat. Keeping this as line/character to be coherent with LSP